### PR TITLE
update dependencies + add clippy to CI

### PR DESCRIPTION
**Pull request by little-dude**
- [X] update dependencies. One nice thing is that with recent `serde_json` versions, we can use the `json!` marco instead of builders
- [x] fix all the clippy warnings
- [x] run clippy in CI for nightly builds

### CODE REVIEW

File: src/screen.rs [hunk 1]
```diff
@@ -1,3 +1,4 @@
+#![allow(dead_code)]
```
Comment (ajm188) [inline src/screen.rs]: I'd rather keep the warnings on, that we can track down unused functions (example: redraw() can be deleted)

File: src/op.rs [hunk 2]
```diff
@@ -10,20 +11,23 @@ pub enum OpType {
     Ins,
 }
 
-impl OpType {
-    fn from_str(op: &str) -> OpType {
+impl FromStr for OpType {
```
Comment (ajm188) [inline src/op.rs:14]: ++

**Review by ajm188 — CHANGES_REQUESTED** [review]
awesome! just want to remove the added "ignore dead code" (and possibly from other files too? let me know your thoughts)

**Comment by little-dude** [comment]
I remove the `allow[dead_code]` that I added. I agree it's better to try getting rid of the warnings than hide them. I also tried to remove it in `src/core.rs` but it becomes very noisy and it gets hard to see the other warnings, so I left it.

**Comment by ajm188** [comment]
lol fair enough. i'll open an issue for it :)

**Comment by ajm188** [comment]
I'm assuming I should wait for you to get clippy working in CI before approving? (I see that box is still unchecked all the way at the top)

**Comment by little-dude** [comment]
Yes, I'm on it right now  actually :)

**Review by ajm188 — APPROVED** [review]

**Pull request merged**